### PR TITLE
Global code update : namespace and src folder

### PR DESCRIPTION
**Pull request by btry**
Update of the plugin to use src and namespace. Some minor fixes.

### CODE REVIEW

File: front/config.php [hunk 1]
```diff
@@ -37,8 +37,7 @@
 //header("Location:../../central.php");
 
 // Entry menu case
-define('GLPI_ROOT', '../..');
-include (GLPI_ROOT . "/inc/includes.php");
+include ("../../../inc/includes.php");
```
Comment (btry) [inline front/config.php:40]: This file must move in front/ folder or GLPI compute a broken relative path to find a twig template

File: src/RuleTest.php [hunk 2]
```diff
@@ -44,10 +48,10 @@
 *   - actions
 *
 **/
-class PluginExampleRuleTest extends Rule {
+class RuleTest extends Rule {
 
    // From Rule
-   static public $right     = 'rule_import';
+   public $right     = 'rule_import';
```
Comment (btry) [inline src/RuleTest.php:54]: CommonDBTM defines this property as not static

File: src/RuleTestCollection.php [hunk 3]
```diff
@@ -32,16 +32,20 @@
...
 }
 
 
-class PluginExampleRuleTestCollection extends RuleCollection {
+class RuleTestCollection extends RuleCollection {
 
    // From RuleCollection
    public $stop_on_first_match = true;
-   static public  $right       = 'rule_import';
+   public $right               = 'rule_import';
```
Comment (btry) [inline src/RuleTestCollection.php:48]: CommonDBTM defines this property as not static

Comment (cedric-anne) [inline src/RuleTestCollection.php:48]: Should probably be
```suggestion
   public static $rightname = 'rule_import';
```

Comment (btry) [reply src/RuleTestCollection.php]: var renamed as submitted [here](https://github.com/pluginsGLPI/example/pull/38#discussion_r926381095) for RuleTestCollection

File: src/RuleTest.php (hunk 2 above)
Comment (btry) [reply src/RuleTest.php]: var renamed as submitted [here](https://github.com/pluginsGLPI/example/pull/38#discussion_r926381095) for RuleTestCollection

File: hook.php [hunk 4]
```diff
@@ -272,7 +276,7 @@ function plugin_example_MassiveActions($type) {
    switch ($type) {
       // New action for core and other plugin types : name = plugin_PLUGINNAME_actionname
       case 'Computer' :
-         return ['PluginExampleExample'.MassiveAction::CLASS_ACTION_SEPARATOR.'DoIt' =>
+         return ['GlpiPlugin\Example\Example'.MassiveAction::CLASS_ACTION_SEPARATOR.'DoIt' =>
```
Comment (orthagh) [inline hook.php:279]: on this case, either use double antislash or getType

File: setup.php [hunk 5]
```diff
@@ -185,45 +193,45 @@ function plugin_init_example() {
...
+   $PLUGIN_HOOKS[Hooks::INFOCOM]['example'] = "plugin_example_infocom_hook";
 
    // pre_show and post_show for tabs and items,
-   // see PluginExampleShowtabitem class for implementation explanations
-   $PLUGIN_HOOKS['pre_show_tab']['example']     = ['PluginExampleShowtabitem', 'pre_show_tab'];
-   $PLUGIN_HOOKS['post_show_tab']['example']    = ['PluginExampleShowtabitem', 'post_show_tab'];
-   $PLUGIN_HOOKS['pre_show_item']['example']    = ['PluginExampleShowtabitem', 'pre_show_item'];
-   $PLUGIN_HOOKS['post_show_item']['example']   = ['PluginExampleShowtabitem', 'post_show_item'];
+   // see GlpiPlugin\Example\Showtabitem class for implementation explanations
+   $PLUGIN_HOOKS[Hooks::PRE_SHOW_TAB]['example']     = [Showtabitem::class, 'pre_show_tab'];
```
Comment (orthagh) [inline setup.php:209]: I personaly have preference for class keyword but try to be consistent (`::getType()` or `::class` everwhere)

Comment (btry) [reply setup.php:209]: switching to ::class everywhere, except for expressions like ```$item::getType()```

**Review by cconard96 — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**